File: app/src/main/java/me/tagavari/airmessage/messaging/ChatCreateAction.java
package me.tagavari.airmessage.messaging;

import android.content.Context;

import androidx.annotation.NonNull;

import me.tagavari.airmessage.enums.ConversationItemType;
import me.tagavari.airmessage.R;

/**
 * A message used to signify the creation of a new chat
 */
public class ChatCreateAction extends ConversationAction {
	public ChatCreateAction(long localID, long date) {
		super(localID, -1, null, date);
	}
	
	@Override
	@ConversationItemType
	public int getItemType() {
		return ConversationItemType.chatCreate;
	}
	
	@Override
	public String getMessageDirect(Context context) {
		return context.getResources().getString(R.string.message_conversationcreated);
	}
	
	@Override
	public boolean supportsBuildMessageAsync() {
		return false;
	}
	
	@NonNull
	@Override
	public ChatCreateAction clone() {
		return new ChatCreateAction(getLocalID(), getDate());
	}
}

File: app/src/main/java/me/tagavari/airmessage/messaging/ChatCreateAction.kt
package me.tagavari.airmessage.messaging

import android.content.Context
import me.tagavari.airmessage.R
import me.tagavari.airmessage.enums.ConversationItemType

/**
 * A message used to signify the creation of a new chat
 */
class ChatCreateAction(localID: Long, date: Long) : ConversationAction(localID, -1, null, date) {
	@get:ConversationItemType
	override val itemType: Int
		get() = ConversationItemType.chatCreate
	
	override fun getMessageDirect(context: Context): String {
		return context.resources.getString(R.string.message_conversationcreated)
	}
	
	override val supportsBuildMessageAsync = false
	
	override fun clone(): ChatCreateAction {
		return ChatCreateAction(localID, date)
	}
}